### ISSUE TITLE
Static build broken with toolchains needing -latomic since version 2.17.0

### ISSUE THREAD

**Issue by ffontaine** [randombit/botan#2622]
Static build with toolchains needing `-latomic` (e.g sparc) is broken since version 2.17.0 and commit 88af81b88976d9a1293280f68df597220ab42767 which reverted commit a837dd63f3ac5a9c206863c82b57c679cb72b8b7:

```
/libcrypto.a(threads_pthread.o): in function `CRYPTO_atomic_add':
threads_pthread.c:(.text+0x1a0): undefined reference to `__atomic_is_lock_free'
/home/buildroot/autobuild/run/instance-0/output-1/per-package/botan/host/opt/ext-toolchain/bin/../lib/gcc/sparc-buildroot-linux-uclibc/9.3.0/../../../../sparc-buildroot-linux-uclibc/bin/ld: threads_pthread.c:(.text+0x1b8): undefined reference to `__atomic_fetch_add_4'
```

As reverting commit 88af81b88976d9a1293280f68df597220ab42767 is probably not acceptable, I would advise to use pkg-config to retrieve openssl dependencies or to link with `-latomic` if it is available.

Full build log is available here: http://autobuild.buildroot.org/results/5c0/5c03ee53a34a3cdb409cffcda76e5cc2c723778b/build-end.log

**Comment by randombit** [comment]
Thanks for the report I'll look into a fix for this, sorry about the regression.

**Comment by randombit** [comment]
@ffontaine Can you try the patch in #2623 it should apply cleanly to 2.17. I will backport to 2.x if everything looks ok.

**Comment by ffontaine** [comment]
I confirm that the patch is working. Thanks.

**Comment by randombit** [comment]
Fix in master and release-2 branches, will be included in 2.17.4 but so far this is the only patch in the queue for that release so I'm probably going to sit on it for a while.
